lib/IDE/Refactoring.cpp

```diff
@@ -6194,16 +6194,12 @@ class AsyncConverter : private SourceEntityWalker {
     return Loc;
   }
 
-  /// If the provided SourceLoc has a preceding comment, print it out. Returns
-  /// true if a comment was printed, false otherwise.
-  bool printCommentIfNeeded(SourceLoc Loc, bool AddNewline = false) {
-    auto PrecedingLoc = getLocIncludingPrecedingComment(Loc);
-    if (Loc == PrecedingLoc)
-      return false;
-    if (AddNewline)
-      OS << "\n";
-    OS << CharSourceRange(SM, PrecedingLoc, Loc).str();
-    return true;
+  /// If the provided SourceLoc has a preceding comment, print it out.
+  void printCommentIfNeeded(SourceLoc Loc) {
+    auto Tokens = SF->getAllTokens();
+    auto TokenIter = token_lower_bound(Tokens, Loc);
+    if (TokenIter != Tokens.end() && TokenIter->hasComment())
+      OS << "\n" << TokenIter->getCommentRange().str();
   }
```

Comment:
Could we extract the comment range part into another method (eg. `findCommentRange`) and then use in both `getLocIncludingPrecedingComment` and `printCommentIfNeeded`? It could return an invalid range if there is no previous comment.

LGTM otherwise